testng-core/src/main/java/org/testng/TestClass.java

```diff
@@ -201,6 +219,7 @@ private void initMethods() {
               false,
               xmlTest,
               eachInstance);
+      afterClassConfig.put(instance, Arrays.asList(m_afterClassMethods));
```

Comment:
If available, consider to use List.of() instead. The result is equivalent but the syntax highlights more the intention.

Reply:
Now this is no longer relevant, because we got rid of the arrays itself.